pkg/util/cloudproviders/cloudproviders.go

```diff
@@ -116,3 +117,22 @@ func GetHostAliases(ctx context.Context) []string {
 
 	return util.SortUniqInPlace(aliases)
 }
+
+// GetPublicIPv4 returns the public IPv4 from different providers
+func GetPublicIPv4(ctx context.Context) (string, error) {
+	publicIPFetcher := map[string]func(context.Context) (string, error){
+		"EC2":   ec2.GetPublicIPv4,
+		"GCE":   gce.GetPublicIPv4,
+		"Azure": azure.GetPublicIPv4,
+	}
```

Comment:
We do this for all method of the cloudprovider package. Having the list in the function make it easier to read, especially since we have N different but similar list.